packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -216,6 +193,30 @@ def models_mode(self) -> Optional[str]:
     def is_cadl(self) -> bool:
         return self.options.get("cadl_file", False)
 
+    def add_body_param_type(
+        self,
+        code_model: Dict[str, Any],
+        body_parameter: Dict[str, Any],
+    ):
+        if (
+            body_parameter
+            and body_parameter["type"]["type"] in ("model", "dict", "list")
+            and any(
+                ct for ct in body_parameter.get("contentTypes", []) if JSON_REGEXP.match(ct)
+            )
+            and not body_parameter["type"].get("xmlMetadata")
+            and not any(t for t in ["flattened", "groupedBy"] if body_parameter.get(t))
+        ):
+            origin_type = body_parameter["type"]["type"]
+            is_dpg_model = body_parameter["type"].get("base") == "dpg"
+            body_parameter["type"] = {
+                "type": "combined",
+                "types": [body_parameter["type"], KNOWN_TYPES["binary"]],
+            }
+            if origin_type == "model" and is_dpg_model and self.models_mode == "dpg":
```

Comment:
@msyyc all that needed to change was we needed to make sure if we are generating without models, we don't add another JSON overload